plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeAnalyze.java

```diff
@@ -765,6 +774,109 @@ public void testCollectStatsAfterColumnAdded(boolean collectOnWrite)
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testNoStats()
```

Comment:
Please check also if your logic works for partitioned tables without prior stats

Reply:
Test for partitioned tables added.